csi/moac/node_operator.js

```diff
@@ -119,36 +121,42 @@ class NodeOperator {
     self.eventStream.on('data', async (ev) => {
       if (ev.kind !== 'node') return;
 
-      if (ev.eventType === 'new' || ev.eventType === 'mod') {
-        const name = ev.object.name;
-        const endpoint = ev.object.endpoint;
-        const k8sNode = self.watcher.getRaw(name);
+      await self.workq.push(ev, self._onNodeEvent.bind(self));
+    });
+  }
 
-        if (k8sNode) {
-          // Update object only if it has really changed
-          if (k8sNode.spec.grpcEndpoint !== endpoint) {
-            try {
-              await self._updateResource(name, k8sNode, endpoint);
-            } catch (err) {
-              log.error(`Failed to update node resource "${name}": ${err}`);
-              return;
-            }
-          }
-        } else if (ev.eventType === 'new' && !k8sNode) {
+  async _onNodeEvent (ev) {
+    var self = this;
+    const name = ev.object.name;
+    if (ev.eventType === 'new' || ev.eventType === 'mod') {
+      const endpoint = ev.object.endpoint;
+      const k8sNode = await self.watcher.getRawBypass(name);
+
+      if (k8sNode) {
+        // Update object only if it has really changed
+        if (k8sNode.spec.grpcEndpoint !== endpoint) {
           try {
-            await self._createResource(name, endpoint);
+            await self._updateResource(name, k8sNode, endpoint);
           } catch (err) {
-            log.error(`Failed to create node resource "${name}": ${err}`);
+            log.error(`Failed to update node resource "${name}": ${err}`);
             return;
           }
         }
-        await this._updateStatus(name, ev.object.isSynced() ? 'online' : 'offline');
-      } else if (ev.eventType === 'del') {
-        await self._deleteResource(ev.object.name);
-      } else {
-        assert.strictEqual(ev.eventType, 'sync');
+      } else if (ev.eventType === 'new' && !k8sNode) {
```

Comment:
You don't need the check for '!k8sNode' here because this is already checked by the 'if' on line 135.

Reply:
Done